pyquest_cffi/ops/errors.py

```diff
@@ -785,22 +785,34 @@ def call_interactive(self, qureg: tqureg,
             operators: The Kraus operators
 
         Raises:
+            RuntimeError: Qureg has to be a density matrix qureg but wavefunction qureg was used
+            RuntimeError: Target qubit is not in the Qureg
+            RuntimeError: Too many operators given, must be in [1, 4]
             RuntimeError: Number of target qubits and dimension of Kraus operators mismatch
             RecursionError: Not a valid Kraus map
         """
+        if not qureg.isDensityMatrix:
+            raise RuntimeError("Qureg has to be a density matrix qureg but "
+                               + "wavefunction qureg was used")
+        if qubit not in list(range(qureg.numQubitsRepresented)):
+            raise RuntimeError("Target qubit is not in the Qureg")
+        if len(operators) < 1 or len(operators) > 4:
+            raise RuntimeError("Too many operators given, must be in [1, 4]")
```

Comment:
This will produce a bit of runtime overhead but it is definitely safer. I like it, if we see significant slowdowns we can still revert.